components/board/ArticleList.tsx

```diff
@@ -0,0 +1,65 @@
+import { useEffect, useState } from "react";
+import Pagination from "../common/Pagination";
+import SearchForm from "../common/SearchForm";
+import DropDown from "../common/DropDown";
+import axios from "@/lib/axios";
+import { Order } from "@/types/order";
+import styles from "@/styles/Article.module.scss";
+import { Article } from "@/types/board";
+import Link from "next/link";
+import ArticleCard from "./Article";
+
+export default function ArticleList() {
+  const [articleList, setArticleList] = useState<Article[]>([]);
+  const [order, setOrder] = useState<Order>(Order.Recent);
+  const [keyword, setKeyword] = useState("");
+  const [page, setPage] = useState(1); // 현재 페이지
+  const [pageSize, setPageSize] = useState(10); // 페이지 당 게시글 수
+  const [totalPostCount, setTotalPostCount] = useState(0); // 총 게시글 수
+
+  async function getArticleList() {
+    const res = await axios.get(`/articles`, {
+      params: {
+        page,
+        pageSize,
+        orderBy: order,
+        keyword,
+      },
+    });
```

Comment:
@ssseeo0 

요녀석에 대한 에러 핸들링 케이스가 필요할 것 같습니다.

조금 더 나아가서, `article-list` 라는 콤포넌트가 지닌 목적을 생각해보면 다음과 같은데요
- 글 목록을 조회하기 위한 쿼리 및 페이징 정보 / 콤포넌트 관리
- 렌더링 된 후 글 목록 조회
- 목록된 글에 따라 UI 렌더

이 목적에 따르면 지금 우리가 글 목록을 불러오는 함수는 조금 더 추상화 된 녀석을 사용하는게 좋을 것 같아요.
따라서 다음과 같이 사용할 수 있도록 추상화 시켜주면 어떨까요?
```suggestion
const { data, ...pagingOption } = await getArticlesWithPagination({ page, pageSize, order, keyword })
setArticles(() => data)
setPaginginOption(() => pagingOption)
```